pkg/reconciler/instances/istio/action.go

```diff
@@ -1,410 +1,132 @@
-}
-
-func canUninstall(istioStatus actions.IstioStatus) bool {
-	return isInstalled(istioStatus) && istioStatus.ClientVersion != ""
-}
-
-func getInstalledVersion(context *service.ActionContext, performer actions.IstioPerformer) (actions.IstioStatus, error) {
-	istioStatus, err := performer.Version(context.WorkspaceFactory, context.Task.Version, context.Task.Component, context.KubeClient.Kubeconfig(), context.Logger)
-	if err != nil {
-		return actions.IstioStatus{}, errors.Wrap(err, "Could not fetch Istio version")
-	}
-	context.Logger.Debugf("Detected: istioctl version %s, target Istio version: %s", istioStatus.ClientVersion, istioStatus.TargetVersion)
-	return istioStatus, nil
-}
-
-func isClientCompatibleWithTargetVersion(istioStatus actions.IstioStatus) bool {
-
-	clientHelperVersion, err := newHelperVersionFrom(istioStatus.ClientVersion)
-	if err != nil {
-		return false
-	}
-	targetHelperVersion, err := newHelperVersionFrom(istioStatus.TargetVersion)
-	if err != nil {
-		return false
-	}
-
-	return amongOneMinor(clientHelperVersion, targetHelperVersion)
-}
-
-func canUpdate(istioStatus actions.IstioStatus) (bool, error) {
-	return isComponentCompatible(istioStatus.PilotVersion, istioStatus.TargetVersion, "Pilot")
-}
-
-func ensureCanResetProxies(istioStatus actions.IstioStatus, logger *zap.SugaredLogger) error {
-	pilotVersion, err := istioctl.VersionFromString(istioStatus.PilotVersion)
-	if err != nil {
-		return errors.Wrap(err, "Error parsing pilot version")
+		return err
 	}
 
-	targetVersion, err := istioctl.VersionFromString(istioStatus.TargetVersion)
+	var istioList v1alpha1.IstioList
+	err = k8sClient.List(context.Context, &istioList)
 	if err != nil {
-		return errors.Wrap(err, "Error parsing target version")
-	}
-
-	if pilotVersion.MajorMinorPatch() != targetVersion.MajorMinorPatch() {
-		return fmt.Errorf("Istio pilot version %s do not match target version %s", istioStatus.PilotVersion, istioStatus.TargetVersion)
+		return err
 	}
 
-	for dpVersion := range istioStatus.DataPlaneVersions {
-		if hasDataPlaneOutdatedVersions, _ := isComponentCompatible(dpVersion, istioStatus.TargetVersion, "Data plane"); !hasDataPlaneOutdatedVersions {
-			logger.Warnf("There are service proxies that were not updated during last Istio updates and therefore running on outdated versions.")
-			// We don't want that restarting proxies is blocked if the customer has not restarted their workloads that we cannot restart for them in the proxy reset.
-			// We should try to restart proxies regardless of a version mismatch in the proxies, as we have already verified in this function that the pilot version of Istio
-			// matches the target version and therefore the update of Istio was successful.
-			// Also, we don't want to log the error because it can be confusing because it says that the action is not executed even though it is.
-			return nil
+	if len(istioList.Items) == 0 {
```

Comment:
if there is already an Istio CR on the system in wrong namespace this could be a problem. Could we list only Istio CRs in `kyma-system`, since they matter. Other Istio CRs will have error state and we do not want to break reconciliation for such a case. We do not create Istio CR if there is already one in `kyma-system` namespace.